pkg/url/url.go

```diff
@@ -174,22 +197,49 @@ func Create(client *occlient.Client, kClient *kclient.Client, urlName string, po
 		if err != nil {
 			return "", errors.Wrap(err, "unable to create ingress")
 		}
-		return GetURLString(GetProtocol(routev1.Route{}, *ingress), "", ingressDomain), nil
+		return GetURLString(GetProtocol(routev1.Route{}, *ingress, isExperimental), "", ingressDomain, isExperimental), nil
 	} else {
-		urlName, err := util.NamespaceOpenShiftObject(urlName, applicationName)
-		if err != nil {
-			return "", errors.Wrapf(err, "unable to create namespaced name")
+		if !isRouteSupported {
+			return "", errors.Errorf("routes are not available on non OpenShift clusters")
 		}
-		serviceName, err = util.NamespaceOpenShiftObject(componentName, applicationName)
-		if err != nil {
-			return "", errors.Wrapf(err, "unable to create namespaced name")
+
+		var ownerReference metav1.OwnerReference
+		if !isExperimental {
+			var err error
+			urlName, err = util.NamespaceOpenShiftObject(urlName, applicationName)
+			if err != nil {
+				return "", errors.Wrapf(err, "unable to create namespaced name")
+			}
+			serviceName, err = util.NamespaceOpenShiftObject(componentName, applicationName)
+			if err != nil {
+				return "", errors.Wrapf(err, "unable to create namespaced name")
+			}
+
+			// since the serviceName is same as the DC name, we use that to get the DC
```

Comment:
Is this guaranteed to not change? Asking because if this changes, it might break things, no?

Reply:
Honestly I just copied it from https://github.com/openshift/odo/blob/6caae7ce62cf93adc75083f9a6fd1e66fe4e1250/pkg/occlient/occlient.go#L2614 as now we create the owner reference before calling the function